### ISSUE TITLE
Bug: Broken CRD documentation build

### ISSUE THREAD

**Issue by theunrepentantgeek** [Azure/azure-service-operator#2389]
**Describe the bug**

Our build action for CRD documentation is broken:

```
[controller:gen-crd-docs] generating docs for: ./api/dbformysql/v1beta1
[controller:gen-crd-docs] I0712 04:58:11.577680   48216 main.go:132] parsing go packages in directory ./api/dbformysql/v1beta1
[controller:gen-crd-docs] F0712 04:59:22.346495   48216 main.go:138] no API packages found in ./api/dbformysql/v1beta1 
```

Looks like the documentation tool is looking for something in that folder that doesn't exist, possibly because it's manually written.

**To Reproduce**

Run the GitHub action _Publish API docs_ and review the log output.

**Comment by super-harsh** [comment]
The tool looks for `doc.go` file which our generator puts in each version package. Since its a hand-written resource, our generator did not put `doc.go` file which is what documentation tool is looking for.

**Comment by theunrepentantgeek** [comment]
Fixed in PR #2392
